File: responses/P535GE-KY8/3_longest_common_prefix.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
  table = a[0].split("")
  a.drop(1).each do |string|
    string.split("").each_with_index do |char, index|
      return "" if index == 0 && table[index] != char
      unless table[index] == char
        table = string.split("")[0...index]
        break
      end
    end
  end
  table.join("")
end

File: responses/ZJBAVP-GPH/3_longest_common_prefix.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(X, B) {
    var max = 0;
    
    if (X > B.length) {
        return max;
    }

    for (var i=0; i<(B.length - X), i++) {
        sum = 0;
        for (j=0, j<X, j++) {
            sum += B[j];
        }
        max = Math.max(max)
    }
      
}